homeassistant/components/nest/media_source.py

```diff
@@ -56,6 +60,177 @@
 CLIP_TITLE_FORMAT = "{event_name} @ {event_time}"
 EVENT_MEDIA_API_URL_FORMAT = "/api/nest/event_media/{device_id}/{event_id}"
 
+STORAGE_KEY = "nest.event_media"
+STORAGE_VERSION = 1
+# Write events at most once per duration
+STORAGE_SAVE_DELAY_SECONDS = 5.0
+# Path under config directory
+MEDIA_PATH = f"{DOMAIN}/event_media"
+
+# Size of small in-memory disk cache to avoid excessive disk reads
+DISK_READ_LRU_MAX_SIZE = 32
+
+
+async def async_get_media_event_store(
+    hass: HomeAssistant, subscriber: GoogleNestSubscriber
+) -> EventMediaStore:
+    """Create the disk backed EventMediaStore."""
+    media_path = hass.config.path(MEDIA_PATH)
+
+    def mkdir() -> None:
+        os.makedirs(media_path, exist_ok=True)
+
+    await hass.async_add_executor_job(mkdir)
+    store = Store(hass, STORAGE_VERSION, STORAGE_KEY, private=True)
+    return NestEventMediaStore(hass, subscriber, store, media_path)
+
+
+class NestEventMediaStore(EventMediaStore):
+    """Storage hook to locally persist nest media for events.
+
+    This is interface is meant to provide two storage features:
+    - media storage of events (jpgs, mp4s)
+    - metadata about events (e.g. motion, person), filename of the media, etc.
+
+    The default implementation in nest is in memory, and this allows the data
+    to be backed by disk.
+
+    The nest event media manager internal to the subscriber manages the lifetime
+    of individual objects stored here (e.g. purging when going over storage
+    limits). This store manages the addition/deletion once instructed.
+    """
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        subscriber: GoogleNestSubscriber,
+        store: Store,
+        media_path: str,
+    ) -> None:
+        """Initialize NestEventMediaStore."""
+        self._hass = hass
+        self._subscriber = subscriber
+        self._store = store
+        self._media_path = media_path
+        self._data: dict | None = None
+        self._data_loaded = False
+        self._devices: Mapping[str, str] | None = {}
+
+    async def async_load(self) -> dict | None:
+        """Load data."""
+        if not self._data_loaded:
+            self._devices = await self._get_devices()
+            data = await self._store.async_load()
+            if data is None:
+                _LOGGER.debug("Loaded empty event store")
+                self._data = {}
+            elif isinstance(data, dict):
+                _LOGGER.debug("Loaded event store with %d records", len(data))
+                self._data = data
+            else:
```

Comment:
This case will now log an error but continue, keeping `self._data` as None, causing probably an error later when it expects there to be something ?

Reply:
OK, lets raise ValueError.